// @coinbase/onchainkit/lib/utils/neynar/frame/neynarFrameFunctions.js
'use strict';

const version = require('../../../version.js');
const FetchError = require('../exceptions/FetchError.js');
const neynarFrameModels = require('./neynarFrameModels.js');
const NEYNAR_DEFAULT_API_KEY = 'NEYNAR_ONCHAIN_KIT';
async function neynarFrameValidation(messageBytes, apiKey = NEYNAR_DEFAULT_API_KEY, castReactionContext = true, followContext = true) {
  const options = {
    method: 'POST',
    url: `https://api.neynar.com/v2/farcaster/frame/validate`,
    headers: {
      accept: 'application/json',
      api_key: apiKey,
      'content-type': 'application/json',
      onchainkit_version: version.version
    },
    body: JSON.stringify({
      message_bytes_in_hex: messageBytes,
      cast_reaction_context: castReactionContext,
      // Returns if the user has liked/recasted
      follow_context: followContext // Returns if the user is Following
    })
  };
  const resp = await fetch(options.url, options);
  if (resp.status !== 200) {
    throw new FetchError.FetchError(`non-200 status returned from neynar : ${resp.status}`);
  }
  const responseBody = await resp.json();
  return neynarFrameModels.convertToNeynarResponseModel(responseBody);
}
exports.NEYNAR_DEFAULT_API_KEY = NEYNAR_DEFAULT_API_KEY;
exports.neynarFrameValidation = neynarFrameValidation;
